regparser/web/jobs/views.py

```diff
@@ -1,53 +1,71 @@
-from regparser.web.jobs.models import ParsingJob
-from regparser.web.jobs.serializers import ParsingJobSerializer
+from django.http import HttpResponse
+from regparser.web.jobs.models import (
+    PipelineJob,
+    ProposalPipelineJob,
+    RegulationFile
+)
+from regparser.web.jobs.serializers import (
+    FileUploadSerializer,
+    PipelineJobSerializer,
+    ProposalPipelineJobSerializer
+)
 from regparser.web.jobs.utils import (
     add_redis_data_to_job_data,
     delete_eregs_job,
+    eregs_site_api_url,
+    file_url,
     queue_eregs_job,
+    queue_notification_email,
     status_url
 )
+from rest_framework.parsers import FileUploadParser, MultiPartParser
 from rest_framework.renderers import BrowsableAPIRenderer, JSONRenderer
 from rest_framework.response import Response
 from rest_framework import generics
 from rest_framework import mixins
 from rest_framework import status
 
+import abc
+import hashlib
+
 renderer_classes = (
     JSONRenderer,
     BrowsableAPIRenderer
 )
 
 
-class JobViewList(mixins.ListModelMixin,
-                  mixins.CreateModelMixin,
-                  generics.GenericAPIView):
-    queryset = ParsingJob.objects.all()
-    renderer_classes = renderer_classes
-    serializer_class = ParsingJobSerializer
+class BaseViewList(object):
+    """
+    Intended to be subclassed by classes subclassing ``JobViewList``.
+    Contains the POST-related methods that are relevant to subclasses of
+    ``JobViewList`` but not to ``JobViewList``.
 
-    def filter_queryset(self, request, *args, **kwargs):
-        """
-        Overridden in order to get data from the Redis queue as well as the DB.
+    Should be in the subclass list before ``JobViewList``.
+    """
+    __metaclass__ = abc.ABCMeta
 
-        Impure
-            Pulls information from the DB and the Redis queue.
+    @abc.abstractmethod
+    def build_eregs_args(self, validated_data):
+        """
+        Each type of parser job has its own set of arguments.
+        The ``create`` method calls this method to construct the argument
+        string specific to that type of job.
 
-        :arg HttpRequest request: the incoming request.
+        :arg dict validated_data: Incoming data from the POST that's already
+        been validated by the serializer.
 
-        :rtype: list[ParsingJob]
-        :returns: List of ParsingJob objects.
+        :rtype: list[str]
+        :returns: The components of the argument string in list form.
```

Comment:
It might be that I don't understand the domain well, but I don't know what this means. Would an example be helpful?